aziotd/src/main.rs

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#![deny(rust_2018_idioms, warnings)]
+#![deny(clippy::all, clippy::pedantic)]
+#![allow(clippy::default_trait_access, clippy::let_unit_value)]
+
+mod logging;
+
+#[tokio::main]
+async fn main() {
+    logging::init();
+
+    if let Err(err) = main_inner().await {
+        log::error!("{}", err.0);
+
+        let mut source = std::error::Error::source(&err.0);
+        while let Some(err) = source {
+            log::error!("caused by: {}", err);
+            source = std::error::Error::source(err);
+        }
+
+        log::error!("{:?}", err.1);
+
+        std::process::exit(1);
+    }
+}
+
+async fn main_inner() -> Result<(), Error> {
+    let argv0 = std::env::args_os()
+        .next()
+        .ok_or_else(|| ErrorKind::GetProcessName("argv[0] not set".into()))?;
+
+    // argv[0] could be a single component like "aziot-certd", or a path that ends with "aziot-certd",
```

Comment:
I don't follow. The point is that a simple check for `match argv0 { Some("aziot-certd") => { ... }, ... }` is not sufficient since it might be a full path.

Reply:
yep, sorry, this comment stemmed from the fact that I forgot that we were checking `argv[0]` and not `argv[1]`.